mythtv/libs/libmythupnp/upnputil.cpp

```diff
@@ -60,7 +62,7 @@ QString LookupUDN( const QString &sDeviceType )
 
     sList.removeLast();
     QString sName = "UPnP/UDN/" + sList.last();
-    QString sUDN  = UPnp::GetConfiguration()->GetValue( sName, "" );
+    QString sUDN  = GetMythDB()->GetSetting(sName, "");
```

Comment:
Not a DB setting

Reply:
This was the only setting (UPnP/UDN/MediaRenderer) that was potentially saved.  Should it be saved?

It doesn't feel necessary to me to save this value, and doing so wreaks the read only nature of these UPnP settings.

If it should be saved, then should all of the UPnP settings should be in a separate upnp.xml file to preserve read only access to config.xml by the UPnP code?  (So as to not interfere with other code accessing config.xml.)

Reply:
This is one of the few values that are actually of use. The majority of them probably don't even need to be configurable, and can be hard coded. We are currently undertaking surgical separation of the old webserver. the upnp and ssdp stacks and the config.xml handling code (this last bit I did prior to this PR), and this should be looked at as part of that.

Reply:
Does this need to be saved?

It is probably safe to do so, since XmlConfiguration reads in the entire file when created and this is the only code that looks at that value.

What do you think about my [proposed solution](https://github.com/MythTV/mythtv/pull/435#discussion_r775177963) for these not DB settings?

Reply:
> It is probably safe to do so, since XmlConfiguration reads in the entire file when created and this is the only code that looks at that value.

On a second examination, the setting could be lost in the following scenario:

1. MythContext loads config.xml first.
2. UPnP code loads config.xml.
3. UPnP code sets a UDN and saves it.
4. MythContext has changed database parameters, so it saves its new parameters, which erases the new UDN.

This was also present in the original code.  It is not present in your change; however, you introduced global state without locking.